drafts/0007-RFC-BAG-OPERATORS.md

```diff
@@ -44,279 +44,288 @@ INTERSECT(A, B) = DISTINCT(INTERSECT_ALL(A, B))
 
-TODO: I would like some feedback on this example data. In the real world, we don't always control the shape of our data; but a reader may look at this and think "why isn't this flat if employees is homogenous?". As a database schema, this would be an "Employees" table and a "Department" table with a separate relation table. I would like some feedback on example data shape that would avoid this initial reaction.
+These rules tell us, in the presence of schema, the set operators are valid when the argument relations are union-compatible; and the resultant column descriptors inherit their names and nullability from the column descriptors of the arguments.
+
+- \* — If the statement contains a *CORRESPONDING* clause, then the column descriptors of *T1* and *T2* will be compared by the delcared order of column names in the *CORRESPONDING* list.
+
+## Examples
+
+### Guide
+
+#### Union
+1. Homogenous Bags Permissive Typing
+2. Homogenous Bags Strict Typing
+3. Heterogenous Bags Strict Typing
+4. Heterogenous Bags Permissive Typing
+
+#### Intersect, Except, and Special Cases
+1. Basic Intersect
+2. Basic Except
+3. Singleton bag coercion
+4. NULL and MISSING coercion
+
+### Environment
 
 ```
-{ 
+-- Each bag of employees is *almost* the same! Let's see how we can compose them.
+!add_to_global_env { 
   'hr': { 
     'employees': <<
-      { 'id': 1, 'name': 'Bob Smith',   'title': 'Consultant' }, 
-      { 'id': 2, 'name': 'Susan Smith', 'title': 'Manager' },
-      { 'id': 3, 'name': 'Jane Smith',  'title': 'DEI'}
+      { 'name': 'Bob Smith',   'title': 'Consultant' }, 
+      { 'name': 'Susan Smith', 'title': 'Manager' },
+      { 'name': 'Jane Smith',  'title': 'DEI'}
     >>
   },
   'accounting': { 
     'employees': <<
-      { 'id': 1, 'name': 'Quincy Jones',   'title': 'Director' }, 
-      { 'id': 2, 'name': 'James Early Jones', 'title': 'Manager' },
-      { 'id': 3, 'name': 'Indiana Jones',  'title': 'Intern'}
+      { 'level': 3, 'name': 'Quincy Jones',   'title': 'Director' }, 
+      { 'level': 2, 'name': 'James Earl Jones', 'title': 'Manager' },
+      { 'level': 1, 'name': 'Indiana Jones',  'title': 'Intern'}
     >>
   },
   'engineering': { 
     'employees': <<
-      { 'id': 1, 'name': 'Paul McCharmly',   'title': 'Manager' }, 
-      { 'id': 2, 'name': 'George Parasol', 'title': 'SDE' },
-      { 'id': 3, 'name': 'Ringo Stone',  'title': 'Principal SDE'},
-      { 'id': 4, 'name': 'Eric Lennon',  'title': 'Intern'}
+      { 'title': 'Manager', 'name': 'Paul McCharmly' }, 
+      { 'title': 'SDE', 'name': 'George Parasol' },
+      { 'title': 'Principal SDE', 'name': 'Ringo Stone' },
+      { 'title': 'Intern', 'name': 'Eric Lennon' }
     >>
   }
 } 
 ```
 
-### Union Examples
+### Strict Mode Example Column Descriptors
+
+In strict-typing mode, we can define the *hr.employees, accounting.employees, engineering.employees* relations to have the following schemas.
```

Comment:
Note that this is not a characteristic per se of strict typing--I know I sound like a broken record, but this is important.